src/decode.rs

```diff
@@ -4515,14 +4515,13 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
     // update allocation for loopfilter masks
     if num_sb128 != f.lf.mask_sz {
         freep(&mut f.lf.mask as *mut *mut Av1Filter as *mut c_void);
-        freep(&mut f.lf.level as *mut *mut [u8; 4] as *mut c_void);
+        let _ = mem::take(&mut f.lf.level);
         f.lf.mask =
             malloc(::core::mem::size_of::<Av1Filter>() * num_sb128 as usize) as *mut Av1Filter;
         // over-allocate by 3 bytes since some of the SIMD implementations
         // index this from the level type and can thus over-read by up to 3
-        f.lf.level = malloc(::core::mem::size_of::<[u8; 4]>() * num_sb128 as usize * 32 * 32 + 3)
-            as *mut [u8; 4];
-        if f.lf.mask.is_null() || f.lf.level.is_null() {
+        f.lf.level = vec![[0u8; 4]; num_sb128 as usize * 32 * 32 + 3].into();
```

Comment:
```suggestion
        f.lf.level = vec![[0u8; 4]; num_sb128 as usize * 32 * 32 + 3].into(); // TODO fallible allocation
```

We're ignoring allocation failures now, but in case we need to handle them later as `dav1d` did, we want to mark where they're happening.